nntrainer/tensor/tensor.h

```diff
@@ -1202,9 +1202,12 @@ class Tensor {
    *
    * @param tensors tensors to be concatenated to the first tensor
    * @param axis axis
+   * @param output output tensor to store the result
    * @return Tensor concatenated tensor
+   *
+   * @note  This function should not be used directly. Please use cat() instead.
```

Comment:
I need advice on structure and design here.

`Tensor::concat()` is designed for internal use and should not be used directly like this:
```
Tensor t;
t.concat(tensors, axis);
```

This is because the tensor t calling the concat has no relationship with the input and output tensors. Additionally, directly using concat cannot properly handle errors. So, I'm looking for a way to prevent the use of `concat`.

Currently, concat is used as follows.
1. The public static function `Tensor::cat()` utilizes the tensor member function `concat`.
2. `Tensor::concat()` makes use of a tensor private variable `itensor` to call Float/HalfTensor `concat`.

The issue here is that if concat is made private, `Tensor::cat()` wouldn't be able to utilize concat since it's a static function. Is there a way to restrict the usage of concat to only `Tensor::cat()`?